components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/connector/UserClaimUpdateConfigImplTest.java

```diff
@@ -0,0 +1,186 @@
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+
+import javax.xml.namespace.QName;
+
+import static org.powermock.api.mockito.PowerMockito.mock;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * This class does unit test coverage for UserClaimUpdateConfigImpl class.
+ */
+@PrepareForTest({IdentityConfigParser.class})
+public class UserClaimUpdateConfigImplTest extends PowerMockIdentityBaseTest {
+
+    private UserClaimUpdateConfigImpl userClaimUpdateConfig;
+    private static final String CONNECTOR_NAME = "user-claim-update";
+    private static final String CATEGORY = "Account Management Policies";
+    private static final String FRIENDLY_NAME = "User Claim Update";
+    private static final String SUB_CATEGORY = "DEFAULT";
+    private static final String TENANT_DOMAIN = "carbon.super";
+    private static final String USER_CLAIM_UPDATE_ELEMENT = "UserClaimUpdate";
+    private static final String ENABLE_ELEMENT = "Enable";
+    private static final String CLAIM_ELEMENT = "Claim";
+    private static final String CLAIM_URI = "uri";
+    private static final String VERIFICATION_CODE_ELEMENT = "VerificationCode";
+    private static final String EXPIRY_TIME_ELEMENT = "ExpiryTime";
+    private static final String VERIFICATION_ON_UPDATE_ELEMENT = "VerificationOnUpdate";
+
+    @BeforeTest
+    public void init() {
+
+        userClaimUpdateConfig = new UserClaimUpdateConfigImpl();
+    }
+
+    @Test
+    public void testGetName() {
+
+        assertEquals(userClaimUpdateConfig.getName(), CONNECTOR_NAME);
+    }
+
+    @Test
+    public void testGetFriendlyName() {
+
+        assertEquals(userClaimUpdateConfig.getFriendlyName(), FRIENDLY_NAME);
+    }
+
+    @Test
+    public void testGetCategory() {
+
+        assertEquals(userClaimUpdateConfig.getCategory(), CATEGORY);
+    }
+
+    @Test
+    public void testGetSubCategory() {
+
+        assertEquals(userClaimUpdateConfig.getSubCategory(), SUB_CATEGORY);
+    }
+
+    @Test
+    public void testGetOrder() {
+
+        assertEquals(userClaimUpdateConfig.getOrder(), 0);
+    }
+
+    @Test
+    public void testGetPropertyNameMapping() {
+
+        Map<String, String> nameMappingExpected = new HashMap<>();
+        nameMappingExpected.put(IdentityRecoveryConstants.ConnectorConfig.ENABLE_EMAIL_VERIFICATION_ON_UPDATE,
+                "Enable user email verification on update");
+        nameMappingExpected.put(IdentityRecoveryConstants.ConnectorConfig.EMAIL_VERIFICATION_ON_UPDATE_EXPIRY_TIME,
+                "Email verification on update link expiry time");
+        Map<String, String> nameMapping = userClaimUpdateConfig.getPropertyNameMapping();
+        assertEquals(nameMapping, nameMappingExpected, "Maps are not equal.");
```

Comment:
This compares the number of default properties right? So we can check it by comparing the length of maps. But for the error message shall we say the reason why maps are equal? WDYT?

Like: Maps are not equal because...

Reply:
This compares if the content in the 2 maps are the same